libs/store/src/lib/infrastructure/persistence/domain/state-handler.ts

```diff
@@ -0,0 +1,51 @@
+import 'reflect-metadata';
+import { BaseValue } from '../../persistence/domain/base-classes/base-value';
+import { Type } from '../../../domain/public_api';
+
+export function instanceForType<T>(Type: Type<T>, fromState = {}) {
+    const newState = new Type();
+
+    return new Proxy(newState, new StateHandler(fromState));
+}
+/**
+ *  Proxy wrapper
+ */
+class StateHandler<TModel> {
+    constructor(
+        private targetState?: TModel
+    ) {
+    }
+
+    get(target: any, prop: PropertyKey): any {
+        // Check is an array and wraps a new one proxy
+        if (['[object Object]', '[object Array]']
+            .indexOf(Object.prototype.toString.call(this.targetState[prop])) > -1) {
+            return new Proxy(this.targetState[prop], this);
```

Comment:
Can we really share the same Proxy handler with the new child Proxy?
Asking because `this` handler shares the same `targetState` for all underlying proxies. 
At the same time we use `this.targetState` to make an assignment for each particular proxy which represents different model shapes. 
Does it really work? Just can't get it...

Reply:
I am not sure if this can be the same instance, as proxy does not handle dotted notation, of the Target, key, value is always for the main object